php/libraries/NDB_BVL_Battery.class.inc

```diff
@@ -107,6 +107,7 @@ class NDB_BVL_Battery
      *                              true means ONLY first visit false
      *                              means NOT first visit null can be any
      *                              visit.
+     * FIXME: Someone needs to refactor this code. Bad code smell!
```

Comment:
It would be good to say why. Something like "types cannot be enforced because the bool param abuses the fact that PHP allows setting this value to `null` resulting in 3 code paths from a boolean parameter which should only allow for 2...." etc.

Reply:
@johnsaigle added.